src/text_buffer.rs

```diff
@@ -532,11 +544,23 @@ impl TextBuffer {
 
     pub fn undo(&mut self) -> bool {
         let state = self.history.undo(&mut self.row);
+        if let Some((_, _, _, ongoing)) = state {
+            if !ongoing {
+                self.modified_count -= 1;
+            }
+            self.tmp_modified = false;
+        }
         self.after_undoredo(state)
     }
 
     pub fn redo(&mut self) -> bool {
         let state = self.history.redo(&mut self.row);
+        if let Some((_, _, _, ongoing)) = state {
+            if !ongoing {
```

Comment:
ditto

Reply:
If there were some changes not put into history (edited = true),
these changes are put into history
and history after that are cleared.
So no redo occurred and modified count keeps as is.

On the other hand,
modified count should be incremented.